astropy/coordinates/sky_coordinate.py

```diff
@@ -1696,3 +1696,16 @@ def from_name(cls, name, frame='icrs', parse=False):
             return icrs_sky_coord
         else:
             return icrs_sky_coord.transform_to(frame)
+
+
+def skycoord_equal(sc1, sc2):
```

Comment:
Why not implement this as `SkyCoord.__eq__`?

Reply:
That could be much cleaner.

I don't know the reason for there being an `assert_equal` class method on the `AsdfType` for these astropy objects.  I assume `tables` needs it?  It is unique to astropy, as it is not used or required in the `asdf` package.

Reply:
the `assert_equal` on the tags is for the asdf round-trip serialisation machinery right?

Reply:
I would propose it to be in astropy/tests/helper.pym similarly to e.g. `assert_quantity_allclose`

Given it's only used tests, it may also makes sense to similarly call it `assert_skycoord_equal`

Reply:
It can't go in `astropy/tests/helper.py`.  This PR is to get it out of the testing module it is currently in, as that testing module has a `pytest` import and this function is used by production code.

Reply:
I.e. this is not a new function I'm adding.  I'm just moving it from a test area to a non-test area.

Reply:
Its use of `is_equivalent_frame` pretty much ensures that it only works on `SkyCoord`s, so it makes only sense to move it there - be it as a function or a method.

Reply:
And for a function I've been wondering if there should not be a stricter check on the input types, i.e. `isinstance(SkyCoord)` right at the top, raising a more descriptive exception than the current `AttributeError` - but that might be more elegantly solved as a `SkyCoord.__eq__` as well.

Reply:
I think `__eq__` not being defined for `SkyCoord` is intentional, as it is ambiguous (after discussion with @eteq).  Not only is the tolerance an issue, but equality may mean different things for different purposes.  I.e. are two `SkyCoord` instances with different coordinate frames that map to the same celestial position equal?  I don't want to get into any of that in this PR.

Basically, this is only to remove a run-time dependency on `pytest` by moving a few things around.  That's it.